File: web/src/client/sharedAchievements.ts
export const SHARED_ACHIEVEMENT_HEADER = 'X-Eimir-Shared-Achievement';

export type SharedAchievementKind = 'plan-completed' | 'collection-completed';

export function sharedAchievementKind(
  response: Response,
): SharedAchievementKind | null {
  const value = response.headers.get(SHARED_ACHIEVEMENT_HEADER);
  if (value === 'plan-completed' || value === 'collection-completed') {
    return value;
  }
  return null;
}

File: backend/src/eimir/api/shared_achievements.py
"""Shared Achievement response projection helpers for Web/API consumers."""

from __future__ import annotations

SHARED_ACHIEVEMENT_HEADER = "X-Eimir-Shared-Achievement"
PLAN_COMPLETED_ACHIEVEMENT = "plan-completed"
COLLECTION_COMPLETED_ACHIEVEMENT = "collection-completed"

SHARED_ACHIEVEMENT_RESPONSE_HEADER = {
    "description": (
        "Present only when the server confirms an enabled Shared Achievement "
        "for the successful domain completion."
    ),
    "schema": {
        "type": "string",
        "enum": [PLAN_COMPLETED_ACHIEVEMENT, COLLECTION_COMPLETED_ACHIEVEMENT],
    },
}
